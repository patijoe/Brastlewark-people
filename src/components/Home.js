import React from 'react';
import Filter from './Filter';
import { Link } from 'react-router-dom';

class Home extends React.Component {
  render() {
    const { population, handleFilterName, filterName } = this.props;
    console.log('**', population);

    return (
      <div className="app">
        <Filter handleFilterName={handleFilterName} />

        <ul className="population_list">
          {population
          .filter(item => item.name.toUpperCase().includes(filterName.toUpperCase()))
          .map(item => {
            return (
              <li className="person_item" key={item.id}>
                <Link className="home__link" to={`/person/${item.id}`}>
                  <div className="person__container">
                    <div className="person__img" style={{backgroundImage:`url(${item.thumbnail})`}}></div>
                    <h2 className="person_name">{item.name}</h2>

                  </div>
                </Link>
              </li>
            );
          })
          }
        </ul>
      </div>
    );
  }
}

export default Home;