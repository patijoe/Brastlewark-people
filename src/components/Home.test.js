import React from 'react';
import { create } from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Home from './Home';

const population = [
  {
    id: 9,
    name: 'patri',
    thumbnail: 'https://estaticos.elperiodico.com/resources/jpg/1/3/1554908865931.jpg'
  },
  {
    id: 8,
    name: 'julia',
    thumbnail: 'https://estaticos.elperiodico.com/resources/jpg/1/3/1554908865931.jpg'
  }
];

const filterName = '';

describe('Home component', () => {

  it('matches the snapshot', () => {
    const component = create(
    <Router>
      <Home 
        population={population}
        filterName={filterName}
      />
    </Router>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

});