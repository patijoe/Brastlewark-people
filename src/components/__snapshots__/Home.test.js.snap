// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component matches the snapshot 1`] = `
<div
  className="app"
>
  <div
    className="filterName"
  >
    <label
      className="filter__label"
      htmlFor="filterName"
    >
       Who are you looking for?
    </label>
    <input
      id="filterName"
      type="text"
    />
  </div>
  <hr
    className="row"
  />
  <ul
    className="population__list"
  >
    <li
      className="person__item"
    >
      <a
        className="home__link"
        href="/person/9"
        onClick={[Function]}
      >
        <div
          className="person__img"
          style={
            Object {
              "backgroundImage": "url(https://estaticos.elperiodico.com/resources/jpg/1/3/1554908865931.jpg)",
            }
          }
        />
        <h2
          className="person__name"
        >
          patri
        </h2>
      </a>
    </li>
    <li
      className="person__item"
    >
      <a
        className="home__link"
        href="/person/8"
        onClick={[Function]}
      >
        <div
          className="person__img"
          style={
            Object {
              "backgroundImage": "url(https://estaticos.elperiodico.com/resources/jpg/1/3/1554908865931.jpg)",
            }
          }
        />
        <h2
          className="person__name"
        >
          julia
        </h2>
      </a>
    </li>
  </ul>
</div>
`;
